packages/pluggableWidgets/bar-chart-native/src/components/BarChart.tsx

```diff
@@ -0,0 +1,252 @@
+                                        height={chartDimensions?.height}
+                                        width={chartDimensions?.width}
+                                        padding={style.grid?.padding}
+                                        scale={
+                                            dataTypesResult
+                                                ? { x: getScale(dataTypesResult.x), y: getScale(dataTypesResult.y) }
+                                                : undefined
+                                        }
+                                        style={style.grid}
+                                    >
+                                        <VictoryAxis
+                                            orientation={"bottom"}
+                                            dependentAxis
+                                            style={style.grid?.xAxis}
+                                            {...(firstSeries?.xFormatter
+                                                ? { tickFormat: firstSeries.xFormatter }
+                                                : undefined)}
+                                        />
+                                        <VictoryAxis
+                                            style={style.grid?.yAxis}
+                                            orientation={"left"}
+                                            {...(firstSeries?.yFormatter
+                                                ? { tickFormat: firstSeries.yFormatter }
+                                                : undefined)}
+                                        />
+                                        {groupedOrStacked}
+                                    </VictoryChart>
+                                ) : null}
+                            </View>
+                        </View>
+                        {axisLabelStyles.extractedXAxisLabelStyle.relativePositionGrid === "bottom"
+                            ? xAxisLabelComponent
+                            : null}
+                    </View>
+                    {showLegend ? <Legend style={style} series={series} /> : null}
+                </View>
+            )}
+        </View>
+    );
+}
+
+type DataTypeResult = { x: string; y: string };
+
+function getDataTypes(series: BarChartSeries[]): DataTypeResult | Error | undefined {
+    let dataTypes: DataTypeResult | undefined;
+
+    for (const _series of series) {
+        const { dataPoints } = _series;
+
+        if (dataPoints.length) {
+            const { x, y } = dataPoints[0];
+            const xDataType = typeof x;
+            const yDataType = typeof y;
+
+            if (!dataTypes) {
+                dataTypes = { x: xDataType, y: yDataType };
+            } else if (dataTypes && (dataTypes.x !== xDataType || dataTypes.y !== yDataType)) {
+                return new Error("Data types of data points belonging to different series aren't equal");
```

Comment:
In line 82 you just check the type and return null, whats the necessity here to define an error instead of undefined?

Reply:
The Error is primarily used to conditionally show an error message, line 164, based on type.